provider/cluster_rosa_classic_resource.go

```diff
@@ -616,12 +623,59 @@ func (r *ClusterRosaClassicResource) Create(ctx context.Context,
 	}
 	object = add.Body()
 
+	// Wait till the cluster is ready unless explicitly disabled:
+	if !state.WaitWithTimeout.Unknown && !state.WaitWithTimeout.Null {
+		err = r.waitTillClusterIsReady(ctx, state.WaitWithTimeout.Value, object, r.logger)
+		if err != nil {
+			response.Diagnostics.AddError(
+				"Can't poll cluster state",
+				fmt.Sprintf(
+					"Can't poll state of cluster with identifier '%s': %v",
+					object.ID(), err,
+				),
+			)
+			return
+		}
+
+		r.logger.Info(ctx, "Cluster state is %s", object.State())
+	}
 	// Save the state:
-	populateRosaClassicClusterState(ctx, object, state, r.logger, DefaultHttpClient{})
+	err = populateRosaClassicClusterState(ctx, object, state, r.logger, DefaultHttpClient{})
+	if err != nil {
+		response.Diagnostics.AddError(
+			"Can't populate cluster state",
+			fmt.Sprintf(
+				"Received error %v", err,
+			),
+		)
+		return
+	}
 	diags = response.State.Set(ctx, state)
 	response.Diagnostics.Append(diags...)
 }
 
+func (r *ClusterRosaClassicResource) waitTillClusterIsReady(ctx context.Context, WaitWithTimeout int64, object *cmv1.Cluster, logger logging.Logger) error {
+	ready := object.State() == cmv1.ClusterStateReady
+	timeoutInMinutes := time.Duration(WaitWithTimeout) * time.Minute
+	if WaitWithTimeout > 0 && !ready {
```

Comment:
I don't think it necessary to check the state here, it would be verified in the poll
Also, in case `WaitWithTimeout <=0`, caller might conclude the state is ready while its not

Reply:
changed it